### PR TITLE
Add parameter to execute inference requests at a fixed frequency

### PR DESCRIPTION
**Pull request by boonsino**
### Details:
 - Add parameter to execute inference requests at a fixed frequency

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: samples/cpp/benchmark_app/benchmark_app.hpp [hunk 1]
```diff
@@ -65,6 +65,11 @@ static const char cache_dir_message[] = "Optional. Enables caching of loaded mod
 static const char load_from_file_message[] = "Optional. Loads model from file directly without read_model."
                                              " All CNNNetwork options (like re-shape) will be ignored";
 
+/// @brief message for run frequency
+static const char run_frequency_message[] =
```
Comment (ilya-lavrenov) [inline samples/cpp/benchmark_app/benchmark_app.hpp:69]: in serving it's typically named "requests rate"

File: samples/cpp/benchmark_app/benchmark_app.hpp [hunk 2]
```diff
@@ -307,6 +312,9 @@ DEFINE_string(api, "async", api_message);
 /// @brief Number of infer requests in parallel
 DEFINE_uint64(nireq, 0, infer_requests_count_message);
 
+/// @brief Execute infer requests at a fixed frequency
+DEFINE_double(rfreq, 0, run_frequency_message);
```
Comment (Wovchena) [inline samples/cpp/benchmark_app/benchmark_app.hpp:316]: The default could be infinity. That way you don't need to handle 0 separately.

File: samples/cpp/benchmark_app/main.cpp
Comment (Wovchena) [inline samples/cpp/benchmark_app/main.cpp:1]: What is `-rfreq` for? Limiting FPS of an application which is supposed to provide an upper bound for a possible FPS value is strange.

Comment (Wovchena) [inline samples/cpp/benchmark_app/main.cpp:1]: @nkogteva, what were your reasons against limiting infer requests rate introduction?

**Comment by Wovchena** [comment]
build_jenkins

File: samples/cpp/benchmark_app/benchmark_app.hpp (hunk 2 above)
Comment (boonsino) [reply samples/cpp/benchmark_app/benchmark_app.hpp:316]: If set to INFINITY, then the function `if (FLAGS_rfreq > 0)` will still get executed which is redundant right? Having default as 0 would simply skip that and runs without delay. What do you think?

Comment (Wovchena) [reply samples/cpp/benchmark_app/benchmark_app.hpp:316]: I don't know for sure. You should try. I'd expect sleep_for to have the similar logic incide: if <=0 then return.

File: samples/cpp/benchmark_app/main.cpp
[the remark this answers is not in the document]
Comment (boonsino) [reply samples/cpp/benchmark_app/main.cpp:1]: Changed done. Please review.

[the remark this answers is not in the document]
Comment (boonsino) [reply samples/cpp/benchmark_app/main.cpp:1]: New test added. Please review

[the remark this answers is not in the document]
Comment (boonsino) [reply samples/cpp/benchmark_app/main.cpp:1]: Updated.

File: samples/cpp/benchmark_app/benchmark_app.hpp (hunk 1 above)
Comment (boonsino) [reply samples/cpp/benchmark_app/benchmark_app.hpp]: Name changed. Please check if the new naming matches the OV style

File: samples/cpp/benchmark_app/benchmark_app.hpp [hunk 3]
```diff
@@ -65,6 +65,11 @@ static const char cache_dir_message[] = "Optional. Enables caching of loaded mod
 static const char load_from_file_message[] = "Optional. Loads model from file directly without read_model."
                                              " All CNNNetwork options (like re-shape) will be ignored";
 
+/// @brief message for run frequency
+static const char run_frequency_message[] =
+    "Execute at a fixed frequency. Note if the targeted rate per second cannot be reached, "
+    "the benchmark would start the next run immediately, trying its best to catch up.";
```
[the remark this answers is not in the document]
Comment (boonsino) [reply samples/cpp/benchmark_app/benchmark_app.hpp]: The initial wording causes confusion. Updated with different string. Please review.

File: samples/cpp/benchmark_app/benchmark_app.hpp (hunk 2 above)
Comment (boonsino) [reply samples/cpp/benchmark_app/benchmark_app.hpp]: in C++ sleep_for does have the function to return if value is <= 0. But the `time.sleep()` in python cannot take negative value, so a logic to ensure <=0 is needed. To align both C++ and python style, I will keep this logic remained for now. let me know if there's other thought.

**Review by Wovchena — APPROVED** [review]

**Comment by Wovchena** [comment]
build_jenkins

**Comment by Wovchena** [comment]
build_jenkins

**Comment by Wovchena** [comment]
build_jenkins

**Pull request merged**